src/commands/highlights.rs

```diff
@@ -0,0 +1,88 @@
+use super::*;
+use crate::Arc;
+
+/// add a new highlight for the user
+#[command("add_highlight")]
+#[usage("!add_highlight <word, can have spaces>")]
+pub async fn add_highlight(ctx: &client::Context, msg: &Message, args: Args) -> CommandResult {
+    let args = args.message().trim().to_string();
+    if args.contains(" ") {
+        msg.reply_error(
+            &ctx,
+            "Highlight can't contain a space for implementation/performance reasons",
+        )
+        .await?;
+        return Ok(());
+    } else if args.len() == 0 {
+        msg.reply_error(&ctx, "You must provide a argument. ")
+            .await?;
+        return Ok(());
+    }
+
+    let db: Arc<Db> = ctx.get_db().await;
+    match db.set_highlight(msg.author.id, args.clone()).await {
+        Ok(_) => {
+            msg.reply_success(
+                &ctx,
+                format!("You will be notified whenever someone says {}", args),
+            )
+            .await?
+        }
+        Err(e) => {
+            msg.reply_error(&ctx, format!("The database returned a error: {}", e))
+                .await?
+        }
```

Comment:
But it could also be "Doesn't contain valid Unicode" and then the user immediately knows what he did wrong.  I also doubt that the error could leak sensitive information.